R/deleteThis.R

```diff
@@ -0,0 +1,1697 @@
+  #
+  # Args:
+  # - stateKey: named list (from the analysis json) where each name corresponds to a
+  #             name in the state and the value to options in the options list.
+  #             the named entry 'baseSets' contains a list with placeholders that must be
+  #             replaced in the values of the other lists
+  # - options: named list (from the analysis json) with options to run the analysis with
+  #
+  # Return:
+  # - NULL if no statekey was found or no reusable items were found; otherwise
+  #   the reusable items in the state as they were returned from the analysis
+  #
+  state <- .retrieveState()
+  if (! is.null(state)) {
+    if (! is.null(stateKey)) { # new method, from json
+      if ("baseSets" %in% names(stateKey)) # special placeholders will be used in state items
+        key <- .addBaseSetsState(stateKey)
+      else # no special placeholders in any set of options
+        key <- stateKey
+    } else if ("key" %in% names(attributes(state))) { # old method, from analysis
+      key <- attributes(state)$key
+    } else {
+      return(NULL)
+    }
+    state <- .getStateItems(state=state, options=options, key=key)
+  }
+  return(state)
+}
+
+.addBaseSetsState <- function(stateKey) {
+  # Replaces placeholders in state keys with the options associated with those
+  # placeholders
+  #
+  # Args:
+  # - stateKey: named list (from the analysis json) where each name corresponds to a
+  #             name in the state and the value to options in the options list.
+  #             the named entry 'baseSets' contains a list with placeholders that must be
+  #             replaced in the values of the other lists
+  #
+  # Return:
+  # - the altered state keys, now only containing options from the options list
+  #
+  baseSets <- stateKey[["baseSets"]]
+  stateKey[["baseSets"]] <- NULL
+  key <- lapply(stateKey, function(x) {
+    if (any(names(baseSets) %in% x)) {
+      sets <- baseSets[which(names(baseSets) %in% x)]
+      set <- unlist(sets, use.names=FALSE)
+      x <- x[which(! x %in% names(baseSets))]
+      return(c(x, set))
+    }
+    return(x)
+  })
+  return(key)
+}
+
+callback <- function(results=NULL, progress=NULL) {
+
+  ret <- 0
+
+  if (base::exists(".callbackNative")) {
+
+    if (is.null(results)) {
+      json.string <- "null"
+    } else {
+      json.string <- toJSON(.imgToResults(results))
+    }
+
+    if (is.null(progress)) {
+      progress <- -1
+    } else if (! is.numeric(progress)) {
+      stop("Provide a numeric value to the progress updater")
+    }
+
+    response <- .fromRCPP(".callbackNative", json.string, progress)
+
+    if (is.character(response)) {
+
+      ret <- fromJSON(base::paste("[", response, "]"))[[1]]
+
+    } else {
+
+      ret <- response
+    }
+  }
+
+  ret
+}
+
+.cat <- function(object) {
+
+  cat(toJSON(object))
+
+}
+
+.dataFrameToRowList <- function(df, discard.column.names=FALSE) {
```

Comment:
This one is found in the contingency tables

Reply:
good find! I've copied it over to that module. Shall I also do that with the functions specific to the classical meta-analysis?

Reply:
Yeah, I think functions that have been around for years but are only used in a single module don't need to be in jaspBase. Be sure to update the NAMESPACE too

Reply:
done!